### PR TITLE
Add system test for 'Client.list_metrics'.

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1610 as a base.~~

Note the changes to `gcloud.logging`:  the system test revealed that the API [returns "simple" names for the `LogMetric` resource](https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/projects.metrics#LogMetric) (as documented, but which I misread as returning fully-qualified paths).

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes Rebased after merging #1610 to the `logging-api` branch.  PTAL

File: gcloud/logging/metric.py [hunk 1]
```diff
@@ -106,11 +106,8 @@ def from_api_repr(cls, resource, client):
 
         :rtype: :class:`gcloud.logging.metric.Metric`
         :returns: Metric parsed from ``resource``.
-        :raises: :class:`ValueError` if ``client`` is not ``None`` and the
-                 project from the resource does not agree with the project
-                 from the client.
         """
-        metric_name = _metric_name_from_path(resource['name'], client.project)
+        metric_name = resource['name']
```
Comment (dhermes) [inline gcloud/logging/metric.py]: Why this change? I assume the message

> the `LogMetric` resource does not contain a fully-qualified path, but only the simple name

means that there was an assumption from the docs that didn't turn out to be true in the API?

**Comment by dhermes** [comment]
Side question: Can we turn these on to run in Travis?

File: system_tests/logging_.py [hunk 2]
```diff
@@ -80,3 +80,17 @@ def test_create_metric(self):
         metric.create()
         self.to_delete.append(metric)
         self.assertTrue(metric.exists())
+
+    def test_list_metrics(self):
+        metric = Config.CLIENT.metric(
+            DEFAULT_METRIC_NAME, DEFAULT_FILTER, DEFAULT_DESCRIPTION)
+        self.assertFalse(metric.exists())
+        before_metrics, _ = Config.CLIENT.list_metrics()
+        before_names = set(metric.name for metric in before_metrics)
```
Comment (dhermes) [inline system_tests/logging_.py]: Wouldn't they be unique without using `set`?

File: gcloud/logging/metric.py (hunk 1 above)
Comment (tseaver) [inline gcloud/logging/metric.py]: You are correct.  The APIs are maddeningly inconsistent in how they generate such entries.

**Comment by tseaver** [comment]
> Side question: Can we turn these on to run in Travis?

The `Sink` tests (only #1613, at present) are the ones which will need the `Owner` role for the project.  The others should work fine.

File: system_tests/logging_.py (hunk 2 above)
Comment (tseaver) [inline system_tests/logging_.py]: Prep for lines 95 - 96, below.

File: gcloud/logging/metric.py (hunk 1 above)
Comment (dhermes) [inline gcloud/logging/metric.py]: I'd imagine most backend API teams work independently of one another.

File: system_tests/logging_.py (hunk 2 above)
Comment (dhermes) [inline system_tests/logging_.py]: Yup. I originally thought you could use the existing structures like

```python
delta = [metric for metric in after_metrics if metric not in before_names]
```

but since you've got to grab the name, there is no way getting around creating a new container. SGTM.

**Comment by dhermes** [comment]
LGTM
